### PR TITLE
WT-12751 Remove LSM feature

### PR DESCRIPTION
**Pull request by Sean04**
Remove the LSM feature, including some minimal logic relied on only by LSM. Comments are also reworded to no longer mention LSM. 
There are two follow-up tickets that will be merged independently that involve logic that was depended upon by LSM:
[WT-13814](https://jira.mongodb.org/browse/WT-13814) Remove unordered bulk insert config in open_cursor
[WT-13816](https://jira.mongodb.org/browse/WT-13816) Remove insert flag usage for row-store search in search near

### CODE REVIEW

File: src/schema/schema_drop.c [hunk 1]
```diff
@@ -118,9 +118,7 @@ __drop_table(
      * Open the table so we can drop its column groups and indexes.
      *
      * Ideally we would keep the table locked exclusive across the drop, but for now we rely on the
```
Comment (luke-pearson) [inline src/schema/schema_drop.c:120]: This comment is the one I'd like a tech debt ticket on with a FIXME.

File: src/evict/evict_inline.h [hunk 2]
```diff
@@ -646,9 +646,11 @@ __wt_evict_app_assist_worker_check(
         __wt_atomic_loadv64(&txn_global->current) != __wt_atomic_loadv64(&txn_global->oldest_id));
 
     /*
-     * LSM sets the "ignore cache size" flag when holding the LSM tree lock, in that case, or when
-     * holding the handle list, schema or table locks (which can block checkpoints and eviction),
-     * don't block the thread for eviction.
+     * Don't block the thread for eviction when holding the handle list, schema or table locks
```
Comment (luke-pearson) [inline src/evict/evict_inline.h:649]: Is the added detail intentional? If so, great. Also "session's" -> "sessions", I think?

**Review by luke-pearson — APPROVED** [review]
LGTM, two comments need addressing.

File: src/evict/evict_inline.h (hunk 2 above)
Comment (Sean04) [reply src/evict/evict_inline.h:649]: Yes, this is intentional

>"session's" -> "sessions"
yes you're right

File: dist/api_data.py [hunk 3]
```diff
@@ -1906,8 +1819,7 @@ def __ge__(self, other):
         modified. If true, this option forces the checkpoint''',
         type='boolean'),
     Config('name', '', r'''
-        if set, specify a name for the checkpoint (note that checkpoints including LSM trees
-        may not be named)'''),
+        if set, specify a name for the checkpoint (note that checkpoints may not be named)'''),
```
Comment (lukech) [inline dist/api_data.py:1822]: For my curiosity: are there any other types of trees that may not be named?

File: test/format/format_config.c
Comment (lukech) [inline test/format/format_config.c:1]: This change has been covered by [another PR](https://github.com/wiredtiger/wiredtiger/pull/11278/files#diff-697c4698297db509ea7fd0ead993bc92c12a9e68cb1b6497a2385ac06087ee54).

**Comment by lukech** [comment]
As discussed, can we get the "examples" changes (specifically, `examples/c/ex_data_source.c`) included in this PR as well, as it requires some other changes in the src directory?

File: dist/api_data.py (hunk 3 above)
Comment (Sean04) [reply dist/api_data.py:1822]: Thanks for pointing this one out. It looks like no it was just for LSM trees. [This](https://github.com/wiredtiger/wiredtiger/pull/550) is the commit that introduced the check. I will reword remove this note from the config description, it no longer makes sense.

File: test/format/format_config.c
Comment (Sean04) [reply test/format/format_config.c:1]: Good point, I will revert this change. I originally had it to ensure it compiled.

File: src/schema/schema_drop.c (hunk 1 above)
Comment (Sean04) [reply src/schema/schema_drop.c:120]: Sorry I forgot to link that. Here it is:
[WT-13812 Investigate table locking during session->drop](https://jira.mongodb.org/browse/WT-13812)

File: src/include/schema.h [hunk 4]
```diff
@@ -371,72 +371,3 @@ struct __wt_import_list {
             __wt_writeunlock(session, &__conn->hot_backup_lock);                                   \
         }                                                                                          \
     } while (0)
-
-/*
- * WT_WITHOUT_LOCKS --
- *	Drop the handle, table and/or schema locks, perform an operation,
- *	re-acquire the lock(s).
- */
-#define WT_WITHOUT_LOCKS(session, op)                                                              \
```
Comment (sueloverso) [inline src/include/schema.h:380]: Not sad to see this go.

**Review by sueloverso — APPROVED** [review]
LGTM

File: src/evict/evict_inline.h (hunk 2 above)
Comment (Sean04) [reply src/evict/evict_inline.h:649]: Fixed https://github.com/wiredtiger/wiredtiger/pull/11273/commits/948ac1fd23f1048fc68bf8c464fb6babe5b7a8bb

File: src/schema/schema_drop.c (hunk 1 above)
Comment (Sean04) [reply src/schema/schema_drop.c]: Added https://github.com/wiredtiger/wiredtiger/pull/11273/commits/c3e6b8c4a3b94fb6ce1c4cea08a3a3195a99e8bd

File: dist/api_data.py (hunk 3 above)
Comment (Sean04) [reply dist/api_data.py]: Fixed https://github.com/wiredtiger/wiredtiger/pull/11273/commits/9757a68c4c4b1ad5780d2d1461bedf1d35c650f5

File: test/format/format_config.c
Comment (Sean04) [reply test/format/format_config.c:1]: Fixed [Revert "Remove lsm macro reference in test format"](https://github.com/wiredtiger/wiredtiger/pull/11273/commits/f89bd865e57047c4f107f349e393aeaed8f0154f)

**Review by lukech — APPROVED** [review]
LGTM, thanks!

**Pull request merged**